aquadoggo/src/materializer/tasks/schema.rs

```diff
@@ -1,9 +1,292 @@
 // SPDX-License-Identifier: AGPL-3.0-or-later
 
+use log::debug;
+use p2panda_rs::document::DocumentViewId;
+use p2panda_rs::operation::{AsOperation, AsVerifiedOperation, OperationValue};
+use p2panda_rs::schema::SchemaId;
+use p2panda_rs::storage_provider::traits::OperationStore;
+
 use crate::context::Context;
-use crate::materializer::worker::TaskResult;
+use crate::db::traits::{DocumentStore, SchemaStore};
+use crate::materializer::worker::{TaskError, TaskResult};
 use crate::materializer::TaskInput;
 
-pub async fn schema_task(_context: Context, _input: TaskInput) -> TaskResult<TaskInput> {
-    todo!()
+/// A schema task assembles and stores schemas from their views.
+///
+/// Schema tasks are dispatched whenever a schema definition or schema field definition document
+/// has all its immediate dependencies available in the store. It collects all required views for
+/// the schema, instantiates it and adds it to the schema provider.
+pub async fn schema_task(context: Context, input: TaskInput) -> TaskResult<TaskInput> {
+    debug!("Working on schema task: {}", input);
```

Comment:
@adzialocha the log line already contains the source of this information so it's enough to name the task input being worked I think

> ...tasks::schema] Working on task: <TaskInput -/b9c90d>

Reply:
(we can also keep it more verbose just that's the reason I originally made this a shorter message)

Reply:
Oh right, lol, i never looked at the "source" to the left! Maybe because I'm not used that the file name accidentally covers the name of the function inside of it 🤣